### ISSUE TITLE
Ability to register templates from Page facade

### ISSUE THREAD

**Issue by Nyratas** [whitecube/nova-page#2]
Registering templates without linking them to a specific Route would open the possibility to create "option forms" with cross-page static content, such as cookie banners, header & footer information, and so on.

**Comment by voidgraphics** [comment]
I have an idea: what if those "option forms" (needs a better name, btw) were defined in the config file, in an array, and were loaded automatically by the package?

We could even have a multi-depth array, to be able to specify which routes the option form should be loaded for. Something like:

```php
'options' => [
    '\App\Nova\Options\CookiesBanner' => '*' // load it for all routes,
    '\App\Nova\Options\ContactInfo' => ['home', 'contact']
]
```

I know that we use Middleware to achieve this with Templates, but I think these are a little too specific to work the same way, since you don't want to load all of them for every page.

And another nice feature we could have is a specific artisan command to generate these, and they would automatically add them to the array in addition to creating the right file.

**Comment by Nyratas** [comment]
Issue was closed by mistake.

**Comment by wimurk** [comment]
This would be awesome. I can create widgets or small cards with text without having to create routes and controllers. 

For the home page where there is a lot of text the edit page drives me crazy with the amount of fields. I would like to split it up  and be able to use it on other pages.

**Comment by raphcollective** [comment]
@voidgraphics @Nyratas I could try to take care of this, do you have a more precise idea of how you wanted this to be implemented? How would you deal with the `$current` variable in the Manager for example?

**Comment by Nyratas** [comment]
Option templates have been added.

It is now possible to register option templates (or regular route templates) in ServiceProviders, like this: 

```php
use Whitecube\NovaPage\Pages\Manager;
use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    public function boot(Manager $pages)
    {
        $pages->register('option', 'footer', \App\Nova\Templates\FooterOptions::class);
    }
}
```

This new feature will be shipped in next release, `v0.2.3`.
